CredentialsManager/App.cs

```diff
@@ -0,0 +1,161 @@
+﻿using System;
+using System.Collections.Generic;
+using CredentialsManager.FilesExceptions;
+using CredentialsManager.Models;
+
+namespace CredentialsManager
+{
+  public static class App
+  {
+    private static readonly Menu MainMenu = CreateMenu();
+
+    private const string Header = "Credentials Manager";
+    private const string LoginMessage = "Welcome";
+    private const string FailedLoginMessage = "Invalid credentials!";
+    private const string RegistrationSuccessfulMessage = "Registration successful!";
+
+    private static bool _isExitInvoked;
+
+    public static void Run()
+    {
+      while (!_isExitInvoked)
+      {
+        try
+        {
+          Manager.Initialize();
+          PrintMenu();
+          GetUserInput();
+        }
+        catch (MyFileNotFoundException ex)
+        {
+          Console.WriteLine(ex.Message);
+          break;
+        }
+        catch (DuplicateUserCredentialsException ex)
+        {
+          Console.WriteLine(ex.Message);
+          Pause("Press ENTER to exit..");
+          break;
+        }
+        catch (UserNameIsTakenException ex)
+        {
+          Console.WriteLine(ex.Message);
+          Pause();
+        }
+      }
+    }
+
+    private static IList<MenuItem> GetMenuItems()
+    {
+      return new List<MenuItem>
+      {
+        new MenuItem { ItemId = 1, Label = "Login", Action = Login, ConsoleKey = ConsoleKey.D1 },
```

Comment:
One way I would solve this is to replace `ConsoleKey` with an array or list of the same type and just add both decimal and numpad keys. Calling `Contains` on them should do the trick.
Despite the fact that I could print the index of the `MenuItem` plus one instead of `ItemId` I still don't think it is redundant.

Reply:
I refactored it to use multiple keys here [156c32e](https://github.com/csinn/CSharp-From-Zero-To-Hero-v2/pull/43/commits/156c32e5012146c3d2066fa777438fc3f0372c7d)